src/fetch.ts

```diff
@@ -30,12 +30,12 @@ export async function fetch<T>(
 ): Promise<FetchResponse<T>> {
   let data = undefined as unknown as T;
   try {
-    const resp = await fetchRaw(url, init);
+    const resp = await undiciFetch(url, init);
     const contentType = resp.headers.get("content-type") || "";
 
     // Parse the response as JSON if the content type is JSON
     if (contentType.includes("application/json")) {
-      data = await resp.json();
+      data = (await resp.json()) as T;
```

Comment:
Type-coercion here might be avoided by updating the return type of this function

Reply:
We don't want to change the return type of this function bc it is exposed to the end user in the lib